interpreter/cling/lib/Utils/SourceNormalization.cpp

```diff
@@ -71,14 +76,34 @@ class MinimalPPLexer: public Lexer {
         return false;
     }
 
-    // Function or class name should be in Tok now
-    if (Identifier(Tok).empty())
+    auto LastTok{Tok};
+    // skip scopes in function name
+    // e.g. int ::the_namespace::class_a::mem_func() { return 3; }
+    if ((Tok.is(tok::coloncolon) && !SkipScopes(Tok, LastTok)) ||
+        (!LexClean(Tok) ||
+         (Tok.is(tok::coloncolon) && !SkipScopes(Tok, LastTok))))
       return false;
 
-    // Advance to argument list or method name
-    if (!LexClean(Tok))
+    const auto Ident{Identifier(LastTok)}; // function, operator or class name
+
+    if (Ident.empty())
       return false;
 
+    if (Ident.equals("operator")) {
+      // Tok is the operator, e.g. <=, ==, +...
+      // however, for operator() and [], Tok only contains the
+      // left side so we need to parse the closing right side
+      // TODO: tok::spaceship operator<=>
+      if ((Tok.isOneOf(tok::l_paren, tok::l_square) && !CheckBalance(Tok)) ||
```

Comment:
Actually `operator int() const { ... }` works, as tested in `interpreter/cling/test/Prompt/decls.C` . but `operator int*() const { ... }` fails.
Speaking about follow-up, will root development team improve cling parsing by patches like this PR?
As a user I am wondering if I should just work-around current parsing issues and wait for root/cling to upgrade llvm version to merge cling with clang-repl (maybe not 100% but at least including the lexing and parsing) or should I patch current cling?
If I will patch more, maybe I should patch clang-repl instead of root/cling if the latter has its lexing/parsing parts in maintenance/sunset mode.
@jalopezg-git What is your opinion?

Reply:
Hi @jiangyilism, sorry for the delay in the reply!

In principle, these part of cling is in maintenance mode, and we are only applying minimal patches to it.  The idea in the future is to rebase cling atop clang-repl.  On the other hand, these kind of patches are not required anymore in clang-repl, as top-level statements are directly parsed and have their own AST node.

Please, feel free to re-request review clicking the corresponding button in the GitHub interface (provided that the patchset is ready)!

Cheers.